server/routes/manageConditions/handlers/curfew/curfewRoutes.ts

```diff
@@ -0,0 +1,116 @@
+import { Request, Response } from 'express'
+import LicenceService from '../../../../services/licenceService'
+import ConditionService from '../../../../services/conditionService'
+import { AddAdditionalConditionRequest, Licence } from '../../../../@types/licenceApiClientTypes'
+import { SimpleTime } from '../../types'
+import { User } from '../../../../@types/CvlUserDetails'
+
+export default class CurfewRoutes {
+  constructor(
+    private readonly licenceService: LicenceService,
+    private readonly conditionService: ConditionService
+  ) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { conditionCode } = req.params
+    const { licence } = res.locals
+
+    const additionalCondition = licence.additionalLicenceConditions.find(condition => condition.code === conditionCode)
+    const conditionInstances = licence.additionalLicenceConditions
+      .filter(condition => condition.code === conditionCode)
+      .sort((a, b) => a.id - b.id)
+    const config = await this.conditionService.getAdditionalConditionByCode(conditionCode, licence.version)
+
+    if (!additionalCondition) {
+      return res.redirect(
+        `/licence/create/id/${licence.id}/additional-licence-conditions${
+          req.query?.fromReview ? '?fromReview=true' : ''
+        }`
+      )
+    }
+
+    const formResponses = conditionInstances.reduce((acc, instance, index) => {
+      instance.data.map(conditionData => {
+        const key = index === 0 ? conditionData.field : `${conditionData.field}${index + 1}`
+        acc[key] = conditionData.value
+        return conditionData.value
+      })
+      return acc
+    }, {})
```

Comment:
Instances of the curfew condition are stored as separate conditions in the DB, but all need to be displayed on one page. This reducer takes all instances and creates one object that houses all of the input data in a way that is compatible with the formbuilder components

Reply:
this is a bit mind melty.. did you mean to use a `foreach`? you seem to be not using the resullts of the `map`?

I wonder if a `flatmap` to produce an array of `[key, value]` then an `Object.fromEntries` would be more elegant..